stl/inc/xstring

```diff
@@ -741,41 +741,36 @@ private:
     bool _Matches[256] = {};
 };
 
-template <class _Traits>
+template <class _Traits, bool _Special = _Is_specialization_v<_Traits, char_traits>>
 constexpr size_t _Traits_find_first_of(_In_reads_(_Hay_size) const _Traits_ptr_t<_Traits> _Haystack,
     const size_t _Hay_size, const size_t _Start_at, _In_reads_(_Needle_size) const _Traits_ptr_t<_Traits> _Needle,
-    const size_t _Needle_size, false_type) noexcept {
-    // in [_Haystack, _Haystack + _Hay_size), look for one of [_Needle, _Needle + _Needle_size), at/after _Start_at
-    // general algorithm
-    if (_Needle_size != 0 && _Start_at < _Hay_size) { // room for match, look for it
-        const auto _End = _Haystack + _Hay_size;
-        for (auto _Match_try = _Haystack + _Start_at; _Match_try < _End; ++_Match_try) {
-            if (_Traits::find(_Needle, _Needle_size, *_Match_try)) {
-                return static_cast<size_t>(_Match_try - _Haystack); // found a match
+    const size_t _Needle_size) noexcept {
+    if constexpr (_Special) {
+        // in [_Haystack, _Haystack + _Hay_size), look for one of [_Needle, _Needle + _Needle_size), at/after _Start_at
+        // special case for std::char_traits
+        if (_Needle_size != 0 && _Start_at < _Hay_size) { // room for match, look for it

```

Comment:
This is duplicated in both branches - we should push it out if the `if constexpr`.